### DISCUSSION TITLE
Backup before migration?

### DISCUSSION THREAD

**Discussion by MatsiukMykola** [sqlalchemy/alembic#1412] [Usage Questions]
Are there any additional easy-to-use auto-backup databases before running migration?
in perfect case - with admin panel to time-shift to some time-moment backup

**Comment by zzzeek** [comment]
that's a really open ended question.

are you asking just about different database vendors, like PostgreSQL vs. MariaDB?  are you looking for commercial solutions?    here's a [well known commercial solution for MySQL](https://www.percona.com/mysql/software/percona-xtrabackup) for example, which I only know of because my dayjob has interacted with that company a bit.     

you'd probably get better answers asking on database message boards in general.

**Comment by MatsiukMykola** [comment]
for Postgres, i just think maybe are already implemented simple free solution - just install and work from a box
